src/NServiceBus.Core.Tests/Faults/MoveFaultsToErrorQueueTests.cs

```diff
@@ -12,89 +12,140 @@ namespace NServiceBus.Core.Tests
     [TestFixture]
     public class MoveFaultsToErrorQueueTests
     {
-        [Test]
-        public async Task ShouldForwardToErrorQueueForAllExceptions()
-        {
-            var fakeFaultPipeline = new FakeFaultPipeline();
-            var errorQueueAddress = "error";
-            var behavior = new MoveFaultsToErrorQueueBehavior(
-                new FakeCriticalError(),
-                errorQueueAddress,
-                "public-receive-address");
+        FakeCriticalError criticalError;
+        FakeFaultPipeline faultPipeline;
 
-            var context = CreateContext("someid", fakeFaultPipeline);
+        [SetUp]
+        public void Setup()
+        {
+            faultPipeline = new FakeFaultPipeline();
+            criticalError = new FakeCriticalError();
+        }
+        
+        [TestCase(TransportTransactionMode.None)]
+        [TestCase(TransportTransactionMode.ReceiveOnly)]
+        [TestCase(TransportTransactionMode.SendsAtomicWithReceive)]
+        [TestCase(TransportTransactionMode.TransactionScope)]
+        public async Task ShouldForwardToErrorQueueForAllExceptions(TransportTransactionMode transactionMode)
+        {
+            var behavior = CreateBehavior(transactionMode, "errors-queue");
+            var context = CreateContext("message-id");
 
-            await behavior.Invoke(context, () => { throw new Exception("testex"); });
+            await behavior.Invoke(context, () => { throw new Exception(); });
 
-            Assert.AreEqual(errorQueueAddress, fakeFaultPipeline.Destination);
+            if (transactionMode != TransportTransactionMode.None)
```

Comment:
My worry is that it would result in a lot of duplication of code. Maybe we can extract the method called `SimulateError` that would take `TransactionMode` and execute it once or twice depending on mode?